components/quick_switch_modal/quick_switch_modal.tsx

```diff
@@ -17,33 +17,55 @@ import SuggestionBox from 'components/suggestion/suggestion_box.jsx';
 import SuggestionList from 'components/suggestion/suggestion_list.jsx';
 import SwitchChannelProvider from 'components/suggestion/switch_channel_provider.jsx';
 import SwitchTeamProvider from 'components/suggestion/switch_team_provider.jsx';
-import NoResultsIndicator from 'components/no_results_indicator/no_results_indicator.tsx';
+import NoResultsIndicator from 'components/no_results_indicator/no_results_indicator';
 
 import {NoResultsVariant} from 'components/no_results_indicator/types';
 
 const CHANNEL_MODE = 'channel';
 const TEAM_MODE = 'team';
 
-export default class QuickSwitchModal extends React.PureComponent {
-    static propTypes = {
+type SwitchToChannelResult = {
+    data?: true;
+    error?: true;
+}
+type ProviderSuggestions = {
+    matchedPretext: any,
+    terms: string[],
+    items: any[],
+    component: SwitchChannelProvider|SwitchTeamProvider,
+}
 
-        /**
-         * The function called to hide the modal
-         */
-        onHide: PropTypes.func.isRequired,
+export type Props = {
 
-        /**
-         * Set to show team switcher
-         */
-        showTeamSwitcher: PropTypes.bool,
+    /**
+     * The function called to hide the modal
+     */
+    onHide: () => void;
 
-        actions: PropTypes.shape({
-            joinChannelById: PropTypes.func.isRequired,
-            switchToChannel: PropTypes.func.isRequired,
-        }).isRequired,
-    };
+    /**
+     * Set to show team switcher
+     */
+    showTeamSwitcher: boolean;
+    actions: {
+        joinChannelById: (channelId: string) => Promise<any>;
+        switchToChannel: (channel: Channel) => Promise<SwitchToChannelResult>;
+    }
+}
+
+type State = {
+    text: string;
+    mode: string|null;
```

Comment:
When would mode ever be null? It seems to only ever be a string here. If we need a default value somewhere though, I'd prefer to use the empty string over null.

Reply:
as far as i can see, the handleOnClick method uses javascript's getAttribute method to get an attribute of an HTML element and sets this state afterwards to that value. In case of the element not being present though (for whatever reasons) this function would return null, hence this state could possibly be null.
source: https://developer.mozilla.org/en-US/docs/Web/API/Element/getAttribute
so I'd leave it like that if that's ok? or am I seeing this wrong here?

Reply:
Yeah, that's fine. I didn't realize that we were getting something from the DOM there. That looks like part of some unfinished functionality around being able to switch to teams with this modal, so I'm not surprised it's a little weird then.